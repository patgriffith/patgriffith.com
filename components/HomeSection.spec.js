import { shallowMount } from '@vue/test-utils';
import HomeSection from './HomeSection';

describe('<HomeSection />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(HomeSection, {propsData: {
      data: {
        slug: ''
      }
    }});
  })

  it('should render <HomeSection/> without crashing', () => {
    wrapper.setProps({
     data: {
       title: 'TEST_TITLE',
       intro: 'TEST_INTRO',
       outro: 'TEST_OUTRO'
     }
    })
    expect(wrapper.element).toMatchSnapshot();
  })

  it('should render expandable sections', () => {

    wrapper.setProps({
      data: {
        expands: ['TEST_SECTION_1', 'TEST_SECTION_2']
      }
     })
     expect(wrapper.element).toMatchSnapshot();
  })

})
