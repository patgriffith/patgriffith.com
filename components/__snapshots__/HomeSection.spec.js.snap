// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeSection /> should render <HomeSection/> without crashing 1`] = `
<section
  class="pt-24 md:pt-32 mt-8"
  id=""
>
  <div
    class="container"
  >
    <h2
      class="mx-auto max-w-xl font-bold text-200 leading-1 text-purple-800 mb-4"
    >
      <span>
        
      </span>
    </h2>
     
    <!---->
     
    <!---->
     
    <!---->
  </div>
</section>
`;

exports[`<HomeSection /> should render expandable sections 1`] = `
<section
  class="pt-24 md:pt-32 mt-8"
  id=""
>
  <div
    class="container"
  >
    <h2
      class="mx-auto max-w-xl font-bold text-200 leading-1 text-purple-800 mb-4"
    >
      <span>
        
      </span>
    </h2>
     
    <!---->
     
    <!---->
     
    <!---->
  </div>
</section>
`;
